src/main/java/net/javadiscord/javabot/systems/help/AutoCodeFormatter.java

```diff
@@ -0,0 +1,102 @@
+	private final BotConfig botConfig;
+	private final UserPreferenceService preferenceService;
+
+	/**
+	 * Method responsible for finding a codeblock, if present.
+	 *
+	 * @param event a {@link MessageReceivedEvent}.
+	 * @return a MessageCodeblock instance, holding a startIndex, content and an endIndex
+	 */
+	@Nullable
+	private static AutoCodeFormatter.CodeBlock findCodeblock(@NotNull MessageReceivedEvent event) {
+		String msg = event.getMessage().getContentRaw();
+		int openingBracket = msg.indexOf("{");
+		int closingBracket = msg.lastIndexOf("}");
+		if (closingBracket == -1 || openingBracket == -1) return null;
+		int startIndex = msg.lastIndexOf("\n", openingBracket);
+		int endIndex = msg.indexOf("\n", closingBracket);
+		if (startIndex == -1) startIndex = 0;
+		if (endIndex == -1) endIndex = msg.length();
+		return new CodeBlock(startIndex, endIndex);
+	}
+
+	/**
+	 * called by {@link HelpListener#onMessageReceived(MessageReceivedEvent)} on every message.
+	 * It is worth noting that this class can't register as an event handler itself due to there being no way of
+	 * setting its methods to be strictly called after {@link HelpListener}.
+	 *
+	 * @param event a {@link MessageReceivedEvent}
+	 */
+	public void handleMessageEvent(@Nonnull MessageReceivedEvent event) {
+		if (!event.isFromGuild()) return;
+		if (event.getAuthor().isBot() || event.getMessage().getAuthor().isSystem()) return;
+		if (autoMod.hasSuspiciousLink(event.getMessage()) || autoMod.hasAdvertisingLink(event.getMessage())) return;
+		if (event.isWebhookMessage()) return;
+		if (!event.isFromThread()) return;
+		if (event.getChannel().asThreadChannel().getParentChannel().getIdLong() != botConfig.get(event.getGuild()).getHelpConfig().getHelpForumChannelId()) {
+			return;
+		}
+		if (!Boolean.parseBoolean(preferenceService.getOrCreate(Objects.requireNonNull(event.getMember()).getIdLong(), Preference.FORMAT_UNFORMATTED_CODE).getState())) {
+			return;
+		}
+
+
+		if (event.getMessage().getContentRaw().contains("```")) return; // exit if already contains codeblock
+
+		CodeBlock code = findCodeblock(event);
+		if (code == null) return;
+
+		if (event.getMessage().getMentions().getUsers().isEmpty() && event.getChannel().asThreadChannel().getTotalMessageCount() > 1) {
+			replaceUnformattedCode(event.getMessage().getContentRaw(), code.startIndex(), code.endIndex(), event);
+		} else sendFormatHint(event);
+	}
+
+	private void sendFormatHint(MessageReceivedEvent event) {
+		event.getChannel().sendMessageEmbeds(formatHintEmbed(event.getGuild())).queue();
+	}
+
+	private void replaceUnformattedCode(String msg, int codeStartIndex, int codeEndIndex, MessageReceivedEvent event) {
+		// default case: a "normal", non-ping containing, non first message of a forum-thread containing "{" and "}".
+		// user must also have set their preferences to allow this.
+		if (event.getMessage().getContentRaw().length() > 1992) { // can't exceed discord's char limit
```

Comment:
Why are you using `event.getMessage().getContentRaw()` when that's the content of `msg`?

Reply:
that's fair enough